### ISSUE TITLE
Change code completion [BUG]

### ISSUE THREAD

**Issue by silverbullettruck2001** [cweijan/vscode-mysql#520]
Can you change the code completion logic so that it only suggests code when a non-whitespace character is entered?  Currently if I change a line of code to insert a space the plugin pops up the code completion dialog.  If you don't want to do this across the board, could you at least allow for a setting to control this?  It's frustrating when you have to add a space the dialog comes up and you have to hit the `esc` key to make the dialog go away when there is nothing that should be suggested just by entering a space character.

Here's the scenario:

I am changing this text that has 6 spaces between `select` and `from` to have an additional space between the 4th and 5th space.

`select     from test;` 

to become:
`select      from test;`

When I do this, it does this:
![autocomplete](https://user-images.githubusercontent.com/26559157/159523845-bc11506d-56a8-4b65-9f3b-103f9decbc97.png)

- OS: mac

- Database Type: mysql

**Comment by cweijan** [comment]
I don't understand what's wrong with this, although the prompt pops up when you press space, it doesn't interrupt your typing.

**Comment by silverbullettruck2001** [comment]
Sorry, I should have mentioned this.  The interruption comes when you try to go to another line.  So in my use case, I added a space and I want to go up to the previous line or down to the next line using the arrow keys on the keyboard.  In those cases it doesn't allow me to do that because the code completion box is there, so I have to hit the escape key to get the box to go away in order to go up or down a line,

**Comment by silverbullettruck2001** [comment]
Also, it just seems unnecessary to display the suggestion when all that was entered was a space.  How helpful is it to display a box with an alphabetical list of SQL functions?  It seems like it would be more helpful to prompt after at least one non-whitespace character is entered so that an accurate suggestion could be provided.

**Comment by cweijan** [comment]
You mentioned two scenarios:
1. When you enter a space, you need to use up and down: In fact, this scenario rarely happens, if it happens occasionally, it is not troublesome to press esc
2. The function list that appears after entering a space is meaningless: here is a list of many commonly used mysql functions, which are useful when you need to perform aggregation operations.

I don't think there will be scenes that often, if you say that the prompt pop-up window distracts your attention, it's more likely
persuasive.

**Comment by silverbullettruck2001** [comment]
I do scenario #1 all the time, which is why it is painful for me and that's why I requested it.  I know I have spoke to others that use this in my area and they had the same concern which is why I created the issue.

For #2, I am not saying that the list of functions is meaningless, but it seems like it is not very useful in the scenario where you are just adding a space and then suggesting "AVG", "CAST", etc.  It seems like an arbitrary suggestion.  I think it is very unlikely that anyone would press the spacebar just to bring up a list of functions in alphabetical order.  I would think it is very useful if I pressed the spacebar, then entered the letter "A" for it to popup a suggestion for "AVG".  But to bring it up just because I pressed the spacebar doesn't seem useful.

**Comment by silverbullettruck2001** [comment]
Also, as I originally suggested, if this is desired functionality could you add a configuration option for this so that people that don't want the code suggestions for when you press the spacebar had a way to disable it and make it work for both scenarios?

**Comment by cweijan** [comment]
It really doesn't make sense to have a function list after a space, I will remove it in the next version.
The auto-completion of this extension is only for serving crud SQL, and these all require space-completion. I think you are writing stored procedures, and other tools should be considered when writing stored procedures.

**Comment by silverbullettruck2001** [comment]
Thanks for your work on this, I appreciate the change.  Do you have a suggestion for a different tool that works better for working with stored procedures?

**Comment by cweijan** [comment]
No, I never use stored procedures, in China, stored procedures are considered an outdated technology and no one uses it.

And this is why this extension has so little support for stored procedures.

**Comment by cweijan** [comment]
I'll try to support configuration triggering in the future.

**Comment by silverbullettruck2001** [comment]
What do you mean by configuration triggering?  Also, if you don't use stored procedures, what do you use to encapsulate database logic?  Do you just use an ORM to do data manipulation?

**Comment by cweijan** [comment]
Mean "Configure which keys trigger completion", spaces are included now.
https://github.com/cweijan/vscode-database-client/blob/6c90929c4d5ea4756e4fd5579c3c6d9fb710cc8a/src/service/serviceManager.ts?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L74
In China, it is all at the application level to replace stored procedures, you don't have to change your habits for this

**Comment by cweijan** [comment]
I've released version 4.8.7, now it's configurable in the vscode settings to modify the triggering method.
```json
{
"database-client.complectionTriggerKey":[ " ", ".", ">", "<", "=", "(" ]
}
```
After modifying this setting, you need to restart vscode.

**Comment by silverbullettruck2001** [comment]
I just checked it out, and it works as expected.  Thanks again!
